client/src/templates/Challenges/redux/completion-epic.js

```diff
@@ -203,6 +204,7 @@ export default function completionEpic(action$, state$) {
             ? of(x, allowBlockDonationRequests({ superBlock, block }))
             : of(x)
         ),
+        mergeMap(x => of(x, setRenderStartTime(new Date().getTime()))),
```

Comment:
I might be misunderstanding this, but:

Doesn't it make more sense to set this when the document starts loading? NOT when the previous challenge is submitted.

I could click on dozens of lessons without submitting any. Are we not interested in these load times?

---

Also, to double-check, does this line only fire if the next lesson is in the same superblock?

If it fires any time, then you could end up with odd render times, because:
1) I submit last lesson in block
2) I spend 10 hours deciding which next block to start
3) I click another lesson
4) Render time is >10 hours